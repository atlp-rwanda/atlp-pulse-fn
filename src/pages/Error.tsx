import React from 'react';
import { useNavigate } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import Button from '../components/Buttons';
import NotFoundImg from '../assets/NotFoundImg.svg';
import useDocumentTitle from '../hook/useDocumentTitle';

function Error() {
  useDocumentTitle('Page not found');
  const { t } = useTranslation();
  const navigate = useNavigate();

  const navigateHome = () => {
    navigate('/');
  };

  return (
    <div className="errorImg flex items-center justify-center h-screen w-screen text-center pt-11">
      <div>
        <img
          src={NotFoundImg}
          className="max-w-36 max-h-32 md:max-w-44 md:max-h-40 mt-10 lg:max-w-52 lg:max-h-48 xl:max-w-2xl xl:max-h-96"
          alt="404"
        />
      </div>
      <div className="max-w-screen-md mt-5 text-white">
        <h1 className="font-black text-3xl md:text-4xl lg:text-5xl xl:text-6xl">
          {t('PAGE NOT FOUND')}
        </h1>
        <p className="my-3 px-8 md:px-14 lg:px-0">
          {t('error_page_paragraph')}
        </p>
        <Button
          onClick={navigateHome}
          variant="primary"
          size="lg"
          data-testid="button-back"
          style="mt-2 lg:mt-5 px-8 text-xl font-bold"
        >
          {t('Back Home')}
        </Button>
      </div>
    </div>
  );
}

export default Error;
