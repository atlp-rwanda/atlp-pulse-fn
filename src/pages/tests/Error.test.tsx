/* eslint-disable */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Error from '../Error';

describe('Page Not Found', () => {
  it('Renders the 404 Page', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <Error />
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
