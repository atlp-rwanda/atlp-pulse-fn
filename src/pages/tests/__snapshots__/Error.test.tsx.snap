// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Not Found Renders the 404 Page 1`] = `
<div
  className="errorImg flex items-center justify-center h-screen w-screen text-center pt-11"
>
  <div>
    <img
      alt="404"
      className="max-w-36 max-h-32 md:max-w-44 md:max-h-40 mt-10 lg:max-w-52 lg:max-h-48 xl:max-w-2xl xl:max-h-96"
      src={Object {}}
    />
  </div>
  <div
    className="max-w-screen-md mt-5 text-white"
  >
    <h1
      className="font-black text-3xl md:text-4xl lg:text-5xl xl:text-6xl"
    >
      PAGE NOT FOUND
    </h1>
    <p
      className="my-3 px-8 md:px-14 lg:px-0"
    >
      error_page_paragraph
    </p>
    <button
      className="btn primary lg mt-2 lg:mt-5 px-8 text-xl font-bold"
      data-testid="button-back"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Back Home
    </button>
  </div>
</div>
`;
